### ISSUE TITLE
metapath_2_vec too slow

### ISSUE THREAD

**Issue by 157459387** [pyg-team/pytorch_geometric#7542]
### 🐛 Describe the bug

I tested metapath2vec model as https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/metapath2vec.py
using only cpu and found the train speed is very slow. 
the reason is that the pyg only used at most cpu, but I have 48 cpus, and the pyg can not used all the cpus to train the model.

### Environment

* PyG version:2.3.1
* PyTorch version:1.12.1
* OS:centos 8
* Python version: python 8
* CUDA/cuDNN version: no
* How you installed PyTorch and PyG (`conda`, `pip`, source): pip
* Any other relevant information (*e.g.*, version of `torch-scatter`):
using docker with shm 36G

**Comment by rusty1s** [comment]
Why does PyG only use 3 CPUs on your end? What happens if you increase the number of workers in the data loader routine?

**Comment by 157459387** [comment]
Thanks @rusty1s  for you reply. I increase the the number of workers in the data loader routine to 24 and  the sometimes  cpu runned on about 12,but most of the time only used less than 3. Then i modified some parameters, like batch size,  walklen, context size as the following:
MetaPath2Vec(data.edge_index_dict, embedding_dim=64, metapath=metapath, walk_length=20, 
                     context_size=5, walks_per_node=5, num_negative_samples=5,
                     sparse=True).to(device)

loader = model.loader(batch_size = 1024, shuffle=True, num_workers=24).
Now the  cpus used 7 stable.  So maybe it's not the pyg's reason. I will try continue.
Do you have any suggests on using more cpus?

**Comment by rusty1s** [comment]
Suggestion from my side would be maybe to do some benchmarking to see which operation might be the bottleneck (model vs data loading, which parts of the model are slow, etc). It might well be that embedding access is slow on CPU if you have a large number of nodes.
